substrate/frame/support/src/migrations.rs

```diff
@@ -260,6 +260,107 @@ pub fn migrate_from_pallet_version_to_storage_version<
 	Pallets::migrate(db_weight)
 }
 
+/// Remove all of a pallet's state and re-initializes it to the current in-code storage version.
+///
+/// It uses the multi block migration frame. Hence it is safe to use even on
+/// pallets that contain a lot of storage.
+///
+/// # Parameters
+///
+/// - P: The pallet to resetted as defined in construct runtime
+/// - W: The weight definition for storage access.
+/// - B, G: Optional. Can be used if the pallet needs to be initialized via [`BuildGenesisConfig`].
+///
+/// # Note
+///
+/// The costs to set the optional genesis state are not accounted for. This should be fine as long
+/// as no massive amounts of data are written.
+pub struct ResetPallet<P, W, B = (), G = ()>(PhantomData<(P, W, B, G)>);
+
+impl<P, W, B, G> ResetPallet<P, W, B, G>
+where
+	P: PalletInfoAccess,
+{
+	fn hashed_prefix() -> [u8; 16] {
+		twox_128(P::name().as_bytes())
+	}
+
+	#[cfg(feature = "try-runtime")]
+	fn num_keys() -> u64 {
+		let prefix = Self::hashed_prefix().to_vec();
+		crate::storage::KeyPrefixIterator::new(prefix.clone(), prefix, |_| Ok(())).count() as _
+	}
+}
+
+impl<P, W, B, G> SteppedMigration for ResetPallet<P, W, B, G>
+where
+	P: PalletInfoAccess + OnGenesis,
+	W: Get<RuntimeDbWeight>,
```

Comment:
RuntimeDbWeight only measure the computational time, should we create a benchmark instead that returns the actual weight of removing one key

Reply:
I assumed that deleting would not incur PoV usage but I need to verify. Then I have to come up with a location where to put the benchmark.